nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/model/impl/ComplexProperty.java

```diff
@@ -154,6 +154,20 @@ public Property get(String name) throws PropertyNotFoundException {
         return getChild(field);
     }
 
+    @Override
+    public void set(String name, Property property) throws PropertyNotFoundException {
+        Field field = getType().getField(name);
+        if (field == null) {
+            Property removedProperty = computeRemovedProperty(name);
+            if (removedProperty != null) {
+                removedProperty.set(name, property);
+            }
+            return;
+        }
+        children.put(field.getName().getPrefixedName(), property);
```

Comment:
In the `get` method which calls `getChild` we use `property.getName()` instead of `field.getName().getPrefixedName()` as the key in the `children` map. Can you make this consistent so that we're sure there's no discrepancy between prefixed/unprefixed?